fe/fe-core/src/main/java/com/starrocks/analysis/LiteralExpr.java

```diff
@@ -246,4 +249,110 @@ public <R, C> R accept(AstVisitor<R, C> visitor, C context) {
     public boolean isSelfMonotonic() {
         return true;
     }
+
+    // Parse from binary data, the format follows mysql binary protocal
+    // see https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_binary_resultset.html.
+    public void parseMysqlParam(ByteBuffer data) {
+        throw new StarRocksPlannerException("Not implement parseMysqlParam in derived class. ", ErrorType.INTERNAL_ERROR);
+    }
+
+    public static LiteralExpr parseLiteral(int encode) throws AnalysisException {
+        LiteralExpr literalExpr = MYSQL_LITERAL_TYPE_ENCODE_MAP.get(encode);
+        if (null != literalExpr) {
+            return literalExpr;
+        } else {
+            throw new AnalysisException("unknown mysql type code " + encode);
+        }
+    }
+
+    private static LiteralExpr tinyIntExpr;
+    private static LiteralExpr smallIntExpr;
+    private static LiteralExpr intExpr;
+    private static LiteralExpr bigIntExpr;
+    private static LiteralExpr floatExpr;
+    private static LiteralExpr doubleExpr;
+    private static LiteralExpr decimal32Expr;
+    private static LiteralExpr timeExpr;
+    private static LiteralExpr dateExpr;
+    private static LiteralExpr dateTimeExpr;
+    private static LiteralExpr stringExpr;
+    private static LiteralExpr varcharExpr;
```

Comment:
not thread safe?

Reply:
those all final static variables , won't be modified .

Reply:
I know that Expr is final, but each `parseMysqlParam` will modify the internal value, which is not thread-safe

Reply:
already add clone()